packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx

```diff
@@ -0,0 +1,183 @@
+                    </FilterButton>
+                ) : (
+                    <Group spacing="xs">
+                        <Select
+                            size="xs"
+                            data={availableFieldOptions}
+                            itemComponent={FilterFieldSelectItem}
+                            placeholder="Select field"
+                            searchable
+                            withinPortal={true}
+                            onChange={(value) => {
+                                setIsAddingFilter(false);
+
+                                if (!value) {
+                                    return;
+                                }
+
+                                const field = fields?.find(
+                                    (f) => f.name === value,
+                                );
+
+                                if (!field) {
+                                    showToastError({
+                                        title: 'Error',
+                                        subtitle: 'Field not found',
+                                    });
+                                    return;
+                                }
+
+                                const defaultOperator =
+                                    field.availableOperators[0];
+
+                                if (!defaultOperator) {
+                                    showToastError({
+                                        title: 'Error',
+                                        subtitle:
+                                            'No filter operators available for this field',
+                                    });
+                                    return;
+                                }
+
+                                dispatch(
+                                    addFilter({
+                                        uuid: uuidv4(),
+                                        field: value,
+                                        fieldKind: field.kind,
+                                        fieldType: field.type,
+                                        operator: defaultOperator,
+                                        values: [],
+                                    }),
+                                );
+                            }}
+                        />
+                        <ActionIcon
+                            size="xs"
+                            onClick={() => setIsAddingFilter(false)}
+                        >
+                            <MantineIcon icon={IconX} />
+                        </ActionIcon>
+                    </Group>
+                )}
+                <Flex w="100%" justify="flex-end">
+                    <Button
+                        bg="black"
+                        onClick={handleApply}
+                        sx={(theme) => ({
+                            ':hover': {
+                                backgroundColor: theme.colors.gray[6],
```

Comment:
still unsure about this button. this is yet another custom styled button

Reply:
Yeah true, I'll keep the default button styling for now and we can change in the future if that's a problem

Reply:
yes please. as mentioned above, I am not excited about introducing new styles till we have a proper UI guide book